editorconfig_plugin/gedit3.py

```diff
@@ -44,6 +45,6 @@ def get_document_properties(self, document):
             location = document.get_file().get_location()
             if location:
                 file_uri = location.get_uri()
-                if file_uri.startswith('file:///'):
-                    return self.get_properties_from_filename(file_uri[7:])
+                if file_uri.startswith('file://'):
+                    return self.get_properties_from_filename(unquote(urlparse(file_uri).path))
```

Comment:
Apologies, @michaelrk02, for not checking this out earlier, but I was just reading the [terminal plugin source](https://gitlab.gnome.org/GNOME/gedit-plugins/-/blob/master/plugins/terminal/terminal.py?ref_type=heads#L350-357), and it turns out **this** plugin is using outdated APIs. This function can be written like this, and not require urllib at all:

```python
location = document.get_file().get_location()
if location and location.has_uri_scheme('file'):
    return self.get_properties_from_filename(location.get_path())
```

(`document.get_file()` returns a `GtkSourceFile`, and its `get_location()` method returns a [`GFile`](https://docs.gtk.org/gio/iface.File.html) with access to all of that interface's methods.)